tests/quic/testQuicConnection.nim

```diff
@@ -57,18 +57,17 @@ suite "quic connection":
 
   asyncTest "notifies about id changes":
     let (client, server) = await setupConnection()
-    defer: await client.drop
-    defer: await server.drop
 
-    var newId: ConnectionId
     server.onNewId = proc (id: ConnectionId) =
-      newId = id
+      check id != ConnectionId.default
```

Comment:
This test will now also pass if the callback is never invoked, because the `check` is now inside the callback.

Reply:
Good catch. It seems the callback for the server isn't called as it used to be. In this test, it is called during `setupConnection`. It works if I use it with the `client`, but I'm not sure we should be making those implicit assumptions about when it gets called. What do you think?

Reply:
The `onNewId()` callback should be called at some point between constructing the server object and having completed the handshake. Now unfortunately the `setupConnection` test helper does two things: constructing the server object and exchanging a first datagram. In this test we'd like to 1) construct the server, 2) set the callback, 3) exchange datagrams, 4) wait for the handshake to complete, and 5) check that the callback is invoked.

So I'd be inclined to split up the `setupConnection` helper somehow.

Reply:
You used this first datagram created by the client to create the server https://github.com/vacp2p/nim-quic/blob/485fd65648f015b16279503dd7300a569b9427a4/quic/transport/ngtcp2/native/server.nim#L111

Reply:
I've updated the test to use the procs inside `setupConnection` directly.